test/blocklib/algorithm/fft/fft_common.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef GRAPH_PROTOTYPE_ALGORITHM_FFT_COMMON_HPP
+#define GRAPH_PROTOTYPE_ALGORITHM_FFT_COMMON_HPP
+
+#include "fft_types.hpp"
+#include <algorithm>
+#include <cmath>
+#include <vector>
+
+namespace gr::algorithm {
+
+template<ComplexType T, typename U>
```

Comment:
```cpp

template<typename T, typename U>
requires ComplexType<T> && std::floating_point<U>
void computeMagnitudeSpectrum(const std::vector<T> &fftOut, std::vector<U> &magnitudeSpectrum, std::size_t fftSize, bool outputInDb) {
    if (fftOut.size() != magnitudeSpectrum.size()) {
        throw std::invalid_argument("Input and output vector sizes must match");
    }

    std::ranges::transform(fftOut, magnitudeSpectrum.begin(), [fftSize, outputInDb](const auto &c) {
        const auto mag{ std::hypot(c.real(), c.imag()) * U(2.0) / static_cast<U>(fftSize) };
        return outputInDb ? U(20.0) * std::log10(mag) : mag;
    });
}
```

Reply:
The utput type `U` can be any numeric type not only `floating_point`, at least as it is now. 
magnitudeSpectrum.size can be also less than fft size, for examples in case of real input.